src/AlgoliaSearchCore.js

```diff
@@ -94,7 +94,8 @@ function AlgoliaSearchCore(applicationID, apiKey, opts) {
     });
 
     // no hosts given, compute defaults
-    this.hosts.read = [this.applicationID + '-dsn.algolia.net'].concat(defaultHosts);
+    var mainSuffix = (opts.dsn === false ? '' : '-dsn') + '.algolia.net';
```

Comment:
tried to be as safe as possible here and only disable if it's explicitly `false`, but maybe another strategy works better @vvo

Reply:
opts is always defined in the code so this can't fail. This it the right way to do it.